### PR TITLE
Delete method for TopicsController

### PR DESCRIPTION
**Pull request by brarrow**
#356 

- [x] Delete method

- [x] Tests for method

### CODE REVIEW

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 1]
```diff
@@ -78,7 +78,28 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
...
+            throws NotFoundException {
+        final TopicModelInterface topicModel = this.topicsService.find(id, topicOptions);
+        if (topicModel == null) {
+            throw new NotFoundException();
+        }
+        return topicModel;
+    }
+
+    private TopicModelInterface getTopicModel(Integer id) throws NotFoundException {
+        return this.getTopicModel(id, this.topicsOptionsBuilder.forDelete());
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: do not use for delete in getTopicModel

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 2]
```diff
@@ -78,7 +78,28 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
...
+
+    @DeleteMapping("/{topicId}")
+    public void delete(@PathVariable Integer topicId) throws NotFoundException, ForbiddenException {
+        final TopicModelInterface topicModel = this.getTopicModel(topicId);
+
+        if (!this.acl.hasAccess(topicModel)) {
+            throw new ForbiddenException();
+        }
+
+        this.topicsService.delete(topicModel);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: that would not work, you should provide correct options for delete object

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 1 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: done

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 2 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: done

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 3]
```diff
@@ -78,7 +78,32 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
...
+
+        if (topicModel == null) {
+            throw new NotFoundException();
+        }
+
+        if (!this.acl.hasAccess(this.topicsService.find(topicId, topicsOptionsBuilder.forAuth()))) {
+            throw new ForbiddenException();
+        }
+
+        this.topicsService.delete(topicModel);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: for delete

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 4]
```diff
@@ -78,7 +78,32 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
         }
         return this.topicsMapper.map(topicModel, Topic.class);
     }
-    /**
-     * delete(topicId)
-     */
+
+    @DeleteMapping("/{topicId}")
+    public void delete(@PathVariable Integer topicId) throws NotFoundException, ForbiddenException {
+        final TopicModelInterface topicModel = this.topicsService.find(topicId, topicsOptionsBuilder.forDelete());
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: getTopicModel (for auth)

forDelete

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java [hunk 5]
```diff
@@ -181,4 +184,45 @@ public void testShowForbidden() throws Exception {
                 .andExpect(content().string(""))
                 .andReturn();
     }
+
+    @Test
+    public void testDeleteSuccess() throws Exception {
+        final TopicModelInterface questionModel = this.topicsSupport.getModelFixtureMock(0);
+        when(this.topicsService.find(any(), any())).thenReturn(questionModel);
+        when(this.acl.hasAccess(any(TopicModelInterface.class))).thenReturn(true);
+        final ArgumentCaptor<TopicModelInterface> argumentCaptor = ArgumentCaptor.forClass(TopicModelInterface.class);
```
Comment (malinink) [inline src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java]: remove

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 3 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: done

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 4 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: done

File: src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java (hunk 5 above)
Comment (brarrow) [reply src/test/java/easytests/api/v1/controllers/TopicsControllerTest.java]: done

**Comment by brarrow** [comment]
@malinink Konstantin Aleksandrovich, check this PR, please.

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 6]
```diff
@@ -78,7 +78,34 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
...
     }
-    /**
-     * delete(topicId)
-     */
+
+    @DeleteMapping("/{topicId}")
+    public void delete(@PathVariable Integer topicId) throws NotFoundException, ForbiddenException {
+        final TopicModelInterface topicModelforAuth = getTopicModel(topicId);
+
+        if (topicModelforAuth == null) {
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: лишее, проверка есть в getTopicModel

File: src/main/java/easytests/api/v1/controllers/TopicsController.java [hunk 7]
```diff
@@ -78,7 +78,34 @@ public Topic show(@PathVariable Integer topicId) throws NotFoundException, Forbi
         }
         return this.topicsMapper.map(topicModel, Topic.class);
     }
-    /**
-     * delete(topicId)
-     */
+
+    @DeleteMapping("/{topicId}")
+    public void delete(@PathVariable Integer topicId) throws NotFoundException, ForbiddenException {
+        final TopicModelInterface topicModelforAuth = getTopicModel(topicId);
```
Comment (malinink) [inline src/main/java/easytests/api/v1/controllers/TopicsController.java]: `this.getTopicModel`

**Review by malinink — CHANGES_REQUESTED** [review]

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 6 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: исправлено

File: src/main/java/easytests/api/v1/controllers/TopicsController.java (hunk 7 above)
Comment (brarrow) [reply src/main/java/easytests/api/v1/controllers/TopicsController.java]: исправлено

**Comment by brarrow** [comment]
@malinink Konstantin Aleksandrovich, merge this PR, please.

**Review by malinink — APPROVED** [review]

**Pull request merged**